internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -1551,3 +1616,100 @@ func (r *BMACReconciler) configureBMHForCleaning(ctx context.Context, log logrus
 	}
 	return
 }
+
+// nodeDrainTimedout returns true if the node drain should be skipped due to timout, false if draining should be run
+func nodeDrainTimedout(bmh *bmh_v1alpha1.BareMetalHost) (bool, error) {
+	annotations := bmh.GetAnnotations()
+	timeoutValue, timeoutExists := annotations[BMH_NODE_DRAIN_TIMEOUT_ANNOTATION]
+	startValue, startExists := annotations[BMH_NODE_DRAIN_START_ANNOTATION]
+
+	if !startExists || !timeoutExists {
```

Comment:
If timeoutExists you can just return true, though I don't see where this annotation is being set.